### PR TITLE
SessionTracker tests

### PR DESCRIPTION
**Pull request by Cawllec**
## Goal
Adds unit tests to the SessionTracker.

PR pointed toward AOST branch for now, as it derives from there.

### CODE REVIEW

**Comment by snmaynard** [comment]
Why isnt this based on master?

File: src/SessionTracker.php [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 namespace Bugsnag;
 
+use Exception;
+use InvalidArgumentException;
```
[the remark this answers is not in the document]
Comment (Cawllec) [reply src/SessionTracker.php]: It didn't work fine before, it just wasn't being tested correctly.  If they get raised there's an error as `Bugsnag\InvalidArgumentException` doesn't exist.

[the remark this answers is not in the document]
Comment (Cawllec) [reply src/SessionTracker.php]: And having `Exception` without qualification won't catch any issues.

**Comment by Cawllec** [comment]
Reduced this PR to a simple addition of SessionTracker tests to the master branch with no dependency on AOST

File: tests/SessionTrackerTest.php [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            [
+                $this->callback(function($session) {
+                    return $session['id'] != null &&
+                        strtotime($session['startedAt']) &&
+                        $session['events']['handled'] == 0 &&
+                        $session['events']['unhandled'] == 0;
+                })
+            ],
+            [
+                null
```
Comment (snmaynard) [inline tests/SessionTrackerTest.php]: whats this?

File: tests/SessionTrackerTest.php [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        $this->sessionTracker->setRetryFunction('retry');
+    }
+
+    public function testCanAddCustomStorageFunction()
+    {
+        $storageStub = $this->getMockBuilder(StorageMock::class)
+                         ->setMethods(['store'])
+                         ->getMock();
+
+        $storageStub->expects($this->exactly(5))->method('store')->withConsecutive(
```
Comment (snmaynard) [inline tests/SessionTrackerTest.php]: I dont understand this test. Can we add some comments to clarify?

File: tests/SessionTrackerTest.php (hunk 2 above)
Comment (snmaynard) [reply tests/SessionTrackerTest.php]: Can we add some comments to clarify what this is?

**Comment by Cawllec** [comment]
I've added comments to clarify the calls in some of the stranger looking tests

**Comment by Cawllec** [comment]
Closing this as an orphaned PR at this point

**Pull request closed without merging**